src/batch/src/executor2/join/hash_join.rs

```diff
@@ -94,11 +97,11 @@ impl<K> Default for HashJoinState<K> {
     }
 }
 
-pub(super) struct HashJoinExecutor<K> {
+pub(super) struct HashJoinExecutor2<K> {
     /// Probe side
-    left_child: BoxedExecutor,
+    left_child: BoxedExecutor2,
     /// Build side
-    right_child: BoxedExecutor,
+    right_child: BoxedExecutor2,
     state: HashJoinState<K>,
```

Comment:
In fact, after using stream api, we no longer need this state machine. We just need to  keep `BuildTable` and `ProbeTable<K>` as local varialbe in `do_execute` method.

Reply:
From my simple understanding, the previous control flow `open next close `caused a lot of variables to be bound to the executor and the whole logic was very complicated. So this morning I made a [version o](https://github.com/singularity-data/risingwave/pull/2102)f the risinglight implementation and simplified it

Reply:
Thanks for you contribution, I'll take a review.

Reply:
> `BuildTable` and `ProbeTable<K>` a

I see what you mean, I'll optimize it